cmd/tenant.go

```diff
@@ -0,0 +1,201 @@
+		return nil, err
+	}
+
+	return resp.Payload, nil
+}
+
+func (c tenantCmd) List() ([]*models.V1TenantResponse, error) {
+	annotations, err := genericcli.LabelsToMap(viper.GetStringSlice("annotation"))
+	if err != nil {
+		return nil, err
+	}
+	resp, err := c.client.Tenant().FindTenants(tenantmodel.NewFindTenantsParams().WithBody(&models.V1TenantFindRequest{
+		ID:          viper.GetString("id"),
+		Name:        viper.GetString("name"),
+		Annotations: annotations,
+	}), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.Payload, nil
+}
+
+func (c tenantCmd) Delete(id string) (*models.V1TenantResponse, error) {
+	resp, err := c.client.Tenant().DeleteTenant(tenantmodel.NewDeleteTenantParams().WithID(id), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.Payload, nil
+}
+
+func (c tenantCmd) Create(rq *models.V1TenantCreateRequest) (*models.V1TenantResponse, error) {
+	resp, err := c.client.Tenant().CreateTenant(tenantmodel.NewCreateTenantParams().WithBody(rq), nil)
+	if err != nil {
+		var r *tenantmodel.CreateTenantConflict
+		if errors.As(err, &r) {
+			return nil, genericcli.AlreadyExistsError()
+		}
+		return nil, err
+	}
+
+	return resp.Payload, nil
+}
+
+func (c tenantCmd) Update(rq *models.V1TenantUpdateRequest) (*models.V1TenantResponse, error) {
+	resp, err := c.client.Tenant().GetTenant(tenantmodel.NewGetTenantParams().WithID(rq.Meta.ID), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// FIXME: should not be done by the client, see https://github.com/fi-ts/cloudctl/pull/26
+	rq.Meta.Version = resp.Payload.Meta.Version + 1
+
+	updateResp, err := c.client.Tenant().UpdateTenant(tenantmodel.NewUpdateTenantParams().WithBody(rq), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return updateResp.Payload, nil
+}
+
+func (c tenantCmd) ToCreate(r *models.V1TenantResponse) (*models.V1TenantCreateRequest, error) {
+	return tenantResponseToCreate(r), nil
+}
+
+func (c tenantCmd) ToUpdate(r *models.V1TenantResponse) (*models.V1TenantUpdateRequest, error) {
+	return tenantResponseToUpdate(r), nil
+}
+
+func tenantResponseToCreate(r *models.V1TenantResponse) *models.V1TenantCreateRequest {
+	return &models.V1TenantCreateRequest{
+		Meta: &models.V1Meta{
+			Apiversion:  r.Meta.Apiversion,
+			Kind:        r.Meta.Kind,
+			ID:          r.Meta.ID,
+			Annotations: r.Meta.Annotations,
+			Labels:      r.Meta.Labels,
+			Version:     r.Meta.Version,
+		},
+		Description: r.Description,
+		Name:        r.Name,
+		Quotas:      r.Quotas,
+	}
+}
+
+func tenantResponseToUpdate(r *models.V1TenantResponse) *models.V1TenantUpdateRequest {
+	return &models.V1TenantUpdateRequest{
+		Name: r.Name,
+		Meta: &models.V1Meta{
+			Apiversion:  r.Meta.Apiversion,
+			Kind:        r.Meta.Kind,
+			ID:          r.Meta.ID,
+			Annotations: r.Meta.Annotations,
+			Labels:      r.Meta.Labels,
+			Version:     r.Meta.Version,
+		},
+		Description:   r.Description,
+		IamConfig:     &models.V1IAMConfig{},
+		DefaultQuotas: &models.V1QuotaSet{},
```

Comment:
Is dangerous as an update will include these empty structs overriding existing data.